grammar.js

```diff
@@ -211,7 +211,10 @@ module.exports = grammar({
         )
       ),
 
-    image_name: ($) => repeat1(choice(/[^@:\s\$]+/, $.expansion)),
+    image_name: ($) => seq(
+      choice(/[^@:\s\$-][^@:\s\$]*/, $.expansion),
```

Comment:
This can be simplified, yes? I think it makes the intent slightly more clear

```suggestion
      choice(/[^@:\s\$-]/, $.expansion),
```

Reply:
I was trying to minimize the number of fragments but it doesn't really matter.

Reply:
Is there any benefit to reducing fragment count other than mental complexity? Performance things I'm not aware of maybe?